cgen error - `expr.sort()` result

**V version:** git master
**OS:** Windows 8

**What did you do?**
```v
a := [3,2].sort()
```
**What did you expect to see?**
Successful compilation.
 
**What did you see instead?**
```
C:\Users\CZASTK~1\AppData\Local\Temp\v\sort.tmp.c:7798:16: error: invalid initializer
  array_int a = qsort(new_array_from_c_array(2, 2, sizeof(int), _MOV((int[2]){3, 2})).data, new_array_from_c_array(2, 2, sizeof(int), _MOV((int[2]){3, 2})).len, new_array_from_c_array(2, 2, sizeof(int), _MOV((int[2]){3, 2})).element_size, compare_ints);
                ^~~~~
```
**Comments**
* `qsort` returns void.
* `builtin/array.v` has `pub fn (mut a []int) sort() {`.

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -910,6 +910,8 @@ pub fn (mut c Checker) call_method(mut call_expr ast.CallExpr) table.Type {
 			// in ['clone', 'str'] {
 			call_expr.receiver_type = left_type.to_ptr()
 			// call_expr.return_type = call_expr.receiver_type
+		} else if method_name == 'sort' {
+			call_expr.return_type = table.void_type
 		}
 		return call_expr.return_type
 	} else if left_type_sym.kind == .map && method_name == 'clone' {

--- a/vlib/v/gen/cgen.v
+++ b/vlib/v/gen/cgen.v
@@ -3907,14 +3907,14 @@ fn (mut g Gen) gen_array_map(node ast.CallExpr) {
 // `users.sort(a.age < b.age)`
 fn (mut g Gen) gen_array_sort(node ast.CallExpr) {
 	// println('filter s="$s"')
-	return_sym := g.table.get_type_symbol(node.return_type)
-	if return_sym.kind != .array {
+	rec_sym := g.table.get_type_symbol(node.receiver_type)
+	if rec_sym.kind != .array {
 		println(node.name)
 		println(g.typ(node.receiver_type))
-		// println(sym.kind)
-		verror('sort() requires an array')
+		// println(rec_sym.kind)
+		verror('.sort() is an array method')
 	}
-	info := return_sym.info as table.Array
+	info := rec_sym.info as table.Array
 	// No arguments means we are sorting an array of builtins (e.g. `numbers.sort()`)
 	// The type for the comparison fns is the type of the element itself.
 	mut typ := info.elem_type

